### PR TITLE
Create subs line items through line item api

### PR DESCRIPTION
**Pull request by brendandeere**
The line items api controller also uses OrderContents#add to create new
line items, meaning that nested attributes cannot be passed to that
endpoint for the subcription line items

This adds the after action hook to build the new subscription on the
correct line item if the subscription_line_item key is passed along with
the line item params

### CODE REVIEW

**Comment by brendandeere** [comment]
I should probably add documentation for this

File: spec/controllers/spree/api/line_items_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+      it_behaves_like 'a new line item'
+
+      it 'creates a new subscription line item' do
+        expect { subject }.
+          to change { SolidusSubscriptions::LineItem.count }.
+          from(0).to(1)
+      end
+
+      it 'creates a subscription line item with the correct values' do
```
Comment (brendandeere) [inline spec/controllers/spree/api/line_items_controller_spec.rb]: No value added REMOVE!!!!!!!!!!!!!!!!!!!

File: app/decorators/spree/controllers/api/line_items/create_subscription_line_items.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        :handle_subscription_line_items,
+        only: :create,
+        if: ->{ params[:subscription_line_item] }
+      )
+    end
+
+    private
+
+    def handle_subscription_line_items
+      line_item = @order.line_items.find_by(variant_id: params[:line_item][:variant_id])
```
Comment (Sinetheta) [inline app/decorators/spree/controllers/api/line_items/create_subscription_line_items.rb]: The line item is already available as `@line_item`

Comment (brendandeere) [inline app/decorators/spree/controllers/api/line_items/create_subscription_line_items.rb]: Nice! I will use that instead! good catch

File: docs/api/v1/line_items.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+```js
+  {
+    order_token: '1234'
+    order: {
+      line_items_attributes: [{
+        // line item attributes
+        subscription_line_items_attributes: [{
+          quantity: 1,          // How many to include in the subscription orders
+          max_installments: 12, // How many times to process the subscriptions (null for ad nauseam)
+          interval: 2592000,    // frequency of subscription orders (in seconds)
```
Comment (Sinetheta) [inline docs/api/v1/line_items.md]: This is no longer accurate.

Comment (brendandeere) [inline docs/api/v1/line_items.md]: :+1:

**Comment by brendandeere** [comment]
Taken kevins comments into account

**Pull request merged**